src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  const resultArray = [];
+  const copyState = { ...state };
+
+  for (const el of actions) {
+    switch (el.type) {
+      case 'addProperties':
+        for (const keys in el.extraData) {
+          copyState[keys] = el.extraData[keys];
+        }
+        resultArray.push(Object.assign({}, copyState));
+        break;
+
+      case 'removeProperties':
+        for (const keys of el.keysToRemove) {
+          delete copyState[keys];
+        }
+        resultArray.push(Object.assign({}, copyState));
```

Comment:
'resultArray.push(Object.assign({}, copyState));' is duplicated, so make sense to move it after the switch case block